source_test.go

```diff
@@ -1010,3 +1041,42 @@ func (stubMetaResolver) ResolveImageConfig(ctx context.Context, ref string, opt
 	}
 	return ref, "", dt, nil
 }
+
+func Test_pathHasPrefix(t *testing.T) {
+	type testCase struct {
+		path   string
+		prefix string
+		expect bool
+	}
+	cases := []testCase{
+		{"/foo", "/foobar", false},
+		{"/foo", "/foo", true},
+		{"/foo/", "/foo", true},
+		{"/foo/", "/foo/", true},
+		{"//foo", "/foo", true},
+		{"//foo/", "/foo", true},
+		{"/foo/bar", "/foo", true},
+		{"/foo/bar/", "/foo", true},
+		{"/foo/bar/", "/foo/", true},
+		{"/foo/bar", "/foo/", true},
+		{"/foo/bar", "/bar", false},
+		{"/foo/bar", "/foo/bar/baz", false},
```

Comment:
maybe you could add:
```suggestion
		{"/foo/bar/baz", "/foo/bar", true},
```

Reply:
meant to be an addition, not an edit

Reply:
Added